compiler/rustc_resolve/src/errors.rs

```diff
@@ -1244,3 +1244,25 @@ pub(crate) struct TraitImplMismatch {
     #[label(resolve_trait_impl_mismatch_label_item)]
     pub(crate) trait_item_span: Span,
 }
+
+#[derive(Diagnostic)]
+#[diag(resolve_restriction_relative_2018)]
+pub(crate) struct RestrictionRelative2018 {
+    #[primary_span]
+    pub(crate) span: Span,
+    #[suggestion(code = "crate::{path_str}", applicability = "maybe-incorrect")]
+    pub(crate) path_span: Span,
+    pub(crate) path_str: String,
+}
+
+#[derive(Diagnostic)]
+#[diag(resolve_restriction_ancestor_only, code = E0742)]
```

Comment:
```suggestion
// FIXME(impl_restrictions): The error code talks about visibility, should be changed
// to talk about restrictions more generally.
#[diag(resolve_restriction_ancestor_only, code = E0742)]
```

Reply:
Would you prefer I update the error message in this PR? I can do that quite easily.

Reply:
Talking about an unstable feature in a stably reachable error code seems iffy. Let's not confuse people, yet.